classes/blocks.php

```diff
@@ -312,12 +332,85 @@ public function add_block_at_end_of_default_region($blockname) {
      */
     public function edit_controls($block) {
         if ($this->is_locked_region($block->instance->region) && $this->is_locked_course_category($this->page->category)) {
-            return [];
+            $controls = [];
+
+            $actionurl = $this->page->url->out(false, array('sesskey' => sesskey()));
+            $blocktitle = $block->title;
+            if (empty($blocktitle)) {
+                $blocktitle = $block->arialabel;
+            }
+
+            if ((get_config('tool_blocksmanager', 'unlockconfig')) && ($this->page->user_can_edit_blocks() || $block->user_can_edit())) {
+                // Edit config icon - always show - needed for positioning UI.
+                $str = new \lang_string('configureblock', 'block', $blocktitle);
+                $controls[] = new \action_menu_link_secondary(
+                    new moodle_url($actionurl, array('bui_editid' => $block->instance->id)),
+                    new \pix_icon('t/edit', $str, 'moodle', array('class' => 'iconsmall', 'title' => '')),
+                    $str,
+                    array('class' => 'editing_edit')
+                );
+            }
+
+            if ($this->can_change_visibility($block)) {
+                // Show/hide icon.
+                if ($block->instance->visible) {
+                    $str = new \lang_string('hideblock', 'block', $blocktitle);
+                    $url = new moodle_url($actionurl, array('bui_hideid' => $block->instance->id));
+                    $icon = new \pix_icon('t/hide', $str, 'moodle', array('class' => 'iconsmall', 'title' => ''));
+                    $attributes = array('class' => 'editing_hide');
+                } else {
+                    $str = new \lang_string('showblock', 'block', $blocktitle);
+                    $url = new moodle_url($actionurl, array('bui_showid' => $block->instance->id));
+                    $icon = new \pix_icon('t/show', $str, 'moodle', array('class' => 'iconsmall', 'title' => ''));
+                    $attributes = array('class' => 'editing_show');
+                }
+                $controls[] = new \action_menu_link_secondary($url, $icon, $str, $attributes);
+            }
+
+            return $controls;
         }
 
         return parent::edit_controls($block);
     }
 
+    /**
+     * Handle showing or hiding a block.
+     * @return boolean true if anything was done. False if not.
+     */
+    public function process_url_show_hide() {
+        if ($blockid = optional_param('bui_hideid', null, PARAM_INT)) {
+            $newvisibility = 0;
```

Comment:
It seems the variable is not used in this function

Reply:
yeps. I was tying to copy the code from the original method without changing it.